core/src/main/java/org/apache/calcite/sql/SqlOperator.java

```diff
@@ -929,6 +929,17 @@ public boolean isDeterministic() {
     return true;
   }
 
+  /**
+   * Returns whether a call to this operator is not sensitive to the operands input order.
+   * An operator is symmetrical if the call returns the same result when
+   * the operands are shuffled.
+   *
+   * <p>By default, returns true for {@link SqlKind#SYMMETRICAL}.
+   */
+  public boolean isSymmetrical() {
+    return SqlKind.SYMMETRICAL.contains(kind);
```

Comment:
The word SYMMETRICAL is very misleading, especially for public method `isSymmetrical`. It doesn't mean order insensitive.
https://www.thefreedictionary.com/Symetrical

Reply:
Yes, we might have a discussion in the mail thread for the name

Reply:
So would you rather believe that Scala guys are always right, even when the dictionary clearly says otherwise?
I do believe you have the common sense of what symmetric encryption algorithm is.
FYI, the original method name of `MurmurHash3#unorderedHash` was `symmetricHash`, it was later [renamed](https://github.com/scala/scala/commit/60fb9ec19b50cd8059122ccffd72014b3eefc51f#diff-1c768a8500fbe02dc18f6333be992081R98) to `unorderedHash`, but the author forgot to update the comment.

Reply:
We might need to discuss on the mail thread about the name, personally i have no strong objections on the name, but i prefer `symmetrical` is better than `inputNonSensitive`, the latter is not even a terminology.

Reply:
Sure, you can send out to the mailing list.

Reply:
I'm siding with @danny0405 on this one. His use conforms to the mathematical concept of a [symmetric function](https://en.wikipedia.org/wiki/Symmetric_function).

Reply:
Well, if you think the meaning the method name `isSymmetrical` conveys is crystal clear and without any ambiguity...

It has several meanings:
- The one you mentioned above, input order insensitive (forgive my ignorance)
- https://www.thefreedictionary.com/Symetrical or https://en.wikipedia.org/wiki/Symmetry_(geometry)
- [Symmetric encryption](https://en.wikipedia.org/wiki/Symmetric-key_algorithm) uses it to denote encryption and decryption use the same key

I do believe there is a reason Scala renamed `symmetricHash` to `unorderedHash`, [Cockroachdb](https://github.com/cockroachdb/cockroach/blob/a6a58f4bd1d6b9a7d9f0de2c690d35b51d4f41ca/pkg/sql/opt/optbuilder/groupby.go#L250) call the method `isOrderingSensitive` instead of others.

For those who lacks of mathematical background, I don't think it is a good naming public API, which is not self explanatory.